### PR TITLE
Convert a personal-graph to and from NetworkX

### LINKED ISSUE #11: Convert a personal-graph to and from NetworkX

### PR DESCRIPTION
**Pull request by anubhuti24**
closes #11

### CODE REVIEW

**Review by sutyum — CHANGES_REQUESTED** [review]

File: personal_graph/ml.py [hunk 1]
```diff
@@ -1,4 +1,73 @@
-"""
-Provide functions to convert the data in the graph db into formats compatible with
-graph analysis frameworks such as Networkx and DGL
-"""
+import json
+import networkx as nx  # type: ignore
+from typing import Dict, Any
+import matplotlib.pyplot as plt
+from personal_graph.graph import Graph
```
[the remark this answers is not in the document]
Comment (sutyum) [reply personal_graph/ml.py:5]: infact the following should work too:
```py
import personal_graph as pg

...

g1 = pg.Graph()

...

g2 = pg.from_networkx(pg.to_nextworkx(g1))

assert g1 == g2, "transforms to and from networkx do not alter the graph"
```

**Review by sutyum — CHANGES_REQUESTED** [review]

File: examples/pythonic_api.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import os
 import logging
 from personal_graph.graph import Graph
+from personal_graph.ml import to_networkx, from_networkx
 from personal_graph.models import Node, EdgeInput, KnowledgeGraph, Edge
```
[the remark this answers is not in the document]
Comment (sutyum) [reply examples/pythonic_api.py:5]: Consider adding the high level imports (models, graph and ml) to the package's root `__init__.py` file in order to support this sort of import ^

File: personal_graph/ml.py [hunk 3]
```diff
@@ -1,4 +1,73 @@
...
+from personal_graph.models import Node, KnowledgeGraph, Edge
+
+
+def to_networkx(graph: Graph, post_visualize: bool = False) -> nx.Graph:
```
[the remark this answers is not in the document]
Comment (anubhuti24) [reply personal_graph/ml.py]: Done

File: personal_graph/ml.py [hunk 4]
```diff
@@ -1,4 +1,73 @@
...
+            font_size=8,
+            node_size=500,
+        )
+        plt.axis("off")
+        plt.show()
+
+    return G
+
+
+def to_personal_graph(network_graph: nx) -> KnowledgeGraph:
```
[the remark this answers is not in the document]
Comment (anubhuti24) [reply personal_graph/ml.py]: Updated

File: personal_graph/ml.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (anubhuti24) [reply personal_graph/ml.py]: Added into examples/pythonic_api.py

File: examples/pythonic_api.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anubhuti24) [reply examples/pythonic_api.py]: Done

File: personal_graph/__init__.py [hunk 5]
```diff
@@ -1,3 +1,15 @@
+from .graph import Graph
 from personal_graph import graph
+from .ml import to_networkx, from_networkx
```
Comment (sutyum) [inline personal_graph/__init__.py:3]: this is an antipattern. Prefer absolute imports: `from personal_graph.ml import to_networkx, from_networkx`

File: personal_graph/__init__.py [hunk 6]
```diff
@@ -1,3 +1,15 @@
+from .graph import Graph
```
Comment (sutyum) [inline personal_graph/__init__.py:1]: easier to just do:
```py
from personal_graph.graph import *
```

while * imports are discouraged elsewhere, they make sense in `__init__.py` files

**Review by sutyum — CHANGES_REQUESTED** [review]

File: personal_graph/__init__.py (hunk 6 above)
Comment (anubhuti24) [reply personal_graph/__init__.py:1]: Alright

Comment (anubhuti24) [reply personal_graph/__init__.py:1]: using * gives mypy error, we can use `from personal_graph.ml import to_networkx, from_networkx` instead.

**Pull request merged**